src/commands/connections/addCosmosDBConnection.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CosmosDBTreeItem } from 'src/explorer/CosmosDBTreeItem';
+import * as vscode from 'vscode';
+import { IAzureNode } from 'vscode-azureextensionui';
+import * as constants from '../../constants';
+import { IConnections } from './IConnections';
+
+export async function addCosmosDBConnection(node: IAzureNode<CosmosDBTreeItem>): Promise<void> {
+    const connectionToAdd = <string>await vscode.commands.executeCommand('cosmosDB.api.getDatabase');
+    if (!connectionToAdd) {
+        return;
+    }
+
+    const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
+    const allConnections = workspaceConfig.get<IConnections[]>(constants.configurationSettings.connections, []);
+    let connectionsUnit = allConnections.find((x: IConnections) => x.webAppId === node.treeItem.client.id);
+    if (!connectionsUnit) {
+        connectionsUnit = <IConnections>{};
+        allConnections.push(connectionsUnit);
+        connectionsUnit.webAppId = node.treeItem.client.id;
+    }
+
+    // tslint:disable-next-line:strict-boolean-expressions
+    connectionsUnit.cosmosDB = connectionsUnit.cosmosDB || [];
+    if (!connectionsUnit.cosmosDB.find((x: string) => x === connectionToAdd)) {
+        connectionsUnit.cosmosDB.push(connectionToAdd);
+        workspaceConfig.update(constants.configurationSettings.connections, allConnections);
+        if (node.treeItem.contextValue === 'AddCosmosDBConnection') {
+            // tslint:disable-next-line:no-non-null-assertion
+            await node.parent!.refresh();
```

Comment:
Is this a good way to resolve the situation when there are no Cosmos DB connections and it provides node as 'Add Cosmos DB Connection' button?

Reply:
Soooo I completely forgot, but you shouldn't be handling refresh yourself. You need to implement the `createChild` method inside `CosmosDBTreeItem`. See here as an example: https://github.com/Microsoft/vscode-azureappservice/blob/master/src/explorer/DeploymentSlotsTreeItem.ts#L59 It will handle refreshing and other UI stuff for you. Once you've implemented that, you just need to call `node.createChild()`

Similarly, the remove code should be implemented in `deleteTreeItem` in `CosmosDBDatabase` tree item. See here for an example: https://github.com/Microsoft/vscode-azureappservice/blob/master/src/explorer/SiteTreeItem.ts#L64

You can fix that stuff now (and let me review the change before merging) or you can merge your existing code as-is and fix in a separate PR.

Reply:
Working on the other thing now, so I'll merge it and will fix it after tomorrow's demo